url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ssamot/config/GameConfig.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *
+ * *** BEGIN LICENSE
+ *  Copyright (C) 2012 Spyridon Samothrakis spyridon.samothrakis@gmail.com
+ *  This program is free software: you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 3, as published
+ *  by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranties of
+ *  MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * *** END LICENSE
+ *
+ */
+
+package ssamot.config;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class GameConfig {
+	private  static String[] configDirectory =  new String[]{"./runtime/","./"};
+	private  int runTimeThreads;
+	
+	
+	private static GameConfig config = null;
+	
+	private double ucb1C = 0.5;
+	private double ucb1MinimumIterations = 1;
+	private String runTimeDirectory= "./";
+
+	
+
+	private GameConfig() {
+		
+	}
+
+	public static GameConfig getInstance() {
+		if(config==null) {
+			Properties prop = new Properties();
+			
+			InputStream is = null;
+			
+			config = new GameConfig();
+			
+			for (int i = 0; i < configDirectory.length; i++) {
+				try {
+					String fileName = configDirectory[i] + "config/main.config";
+					System.out.println("Searching for Config at: " + fileName);
+					is = new FileInputStream(fileName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *is*. Execution paths that do not contain closure statements were detected. Ensure *is* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).